holoviews/plotting/plot.py

```diff
@@ -393,15 +395,66 @@ def _compute_group_range(group, elements, ranges):
 
 
     @classmethod
-    def _deep_options(cls, obj, opt_type, opts, specs=None):
+    def _traverse_options(cls, obj, opt_type, opts,
+                          specs=None, keyfn=None, defaults=False):
         """
         Traverses the supplied object getting all options
-        in opts for the specified opt_type and specs
-        """
-        lookup = lambda x: ((type(x).__name__, x.group, x.label),
-                            {o: cls.lookup_options(x, opt_type).options.get(o, None)
-                             for o in opts})
-        return dict(obj.traverse(lookup, specs))
+        in opts for the specified opt_type and specs. If
+        a keyfn is supplied the returned options will be
+        nested by the returned key. If defaults is True
+        it will also look up the default plot options
+        specified on the matching plot type, appending
+        the default values with ``_default``.
+        """
```

Comment:
Docstring mostly makes sense though I don't get what keyfn is needed for. Also, shouldn't ``defaults=True`` just fill in options that aren't explicitly specified with the defaults instead of adding the '_default' suffix?